python-hadoop/hadoop/io/BytesWritable.py

```diff
@@ -28,4 +28,4 @@ def readFields(self, data_input):
         self._value = data_input.readFully(size)
 
     def toString(self):
-        return ''.join(chr(x) for x in self._value)
+        return ''.join(chr(x) for x in self._value if x > 31 or x == 9)
```

Comment:
I also ran into this problem and this works even better:

    return ''.join(chr(x % 256) for x in self._value)

The main problem is that values are signed short integers (in the range -128 to 127) and `chr` is expecting unsigned integers (in the range 0 to 255).

Would you please confirm that this more general solution works for you too, and update the PR?

Thanks,
Carl